x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeFormatProcessor.java

```diff
@@ -17,55 +18,90 @@
 import java.time.format.DateTimeFormatter;
 import java.time.temporal.TemporalAccessor;
 import java.util.Locale;
+import java.util.Objects;
 
 import static org.elasticsearch.xpack.sql.util.DateUtils.asTimeAtZone;
 
 public class DateTimeFormatProcessor extends BinaryDateTimeProcessor {
 
     public static final String NAME = "dtformat";
+    private static final String[][] JAVA_TIME_FORMAT_REPLACEMENTS = {
+        {"tt", "a"},
```

Comment:
Did you use this table: https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings for the patterns?
I see for example the `F`, `FF`, etc. is not addressed.

Reply:
@matriv I have used same pattern as you mentioned. But i couldn't find any equivalent for `F`,`FF`, ... `FFFFFFF`  format specifiers in Java. Java has some what equivalent specifier which is `S`. Should I proceed with that?

Reply:
I think you can use the square brackets: e.g. `[.SS]` as a translation to `.FF`. Would you mind trying that and add some unit test for it?

Reply:
I think for `[.SS]` is similar to `.SS`. I tried with both and yielding same results. so can I simply use `S` as a replacement of `F` ?

```Java
final Literal dateTime = l(dateTime(2019, 9, 3, 18, 10, 37, Integer.valueOf("001000123")));
new Format(Source.EMPTY, dateTime , l("[SS]:SS"), zoneId).makePipe().asProcessor().process(null)
```
 returns `00:00`

Reply:
Ah, I see. Thanks for checking. I'd say then to just replace `F` with `S` too, otherwise we'd need to use a DateTimeFormatterBuilder and do a more complex parsing of the user provided pattern in order to use something like
`.appendFraction(NANO_OF_SECOND, 0, 9, true)` where 9 should be the number of `F`s provided.
I believe it's not important to implement this, and I'd rather add a NOTE in the docs to clarify that `F` works as `f` (non-optional).

@astefan what do you think?

Reply:
@matriv  Thanks for your suggestion. I have added replacement for `F` with `S` and added note in docs. Let me know if I can mark this conversation as resolved.

Reply:
For me it's fine, let's wait for @astefan 's opinion as well.